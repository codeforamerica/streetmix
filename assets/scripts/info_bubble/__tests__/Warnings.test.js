/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Warnings from '../Warnings'

describe('Warnings', () => {
  it('renders nothing if segment has no warnings', () => {
    const segment = {}
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper.children().length).toBe(0)
  })

  // Note: warnings start indexing at 1!

  it('renders warning 1', () => {
    const segment = {
      warnings: [ null, true ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders warning 2', () => {
    const segment = {
      warnings: [ null, false, true, false ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders warning 3', () => {
    const segment = {
      warnings: [ null, false, false, true ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders two warnings', () => {
    const segment = {
      warnings: [ null, true, false, true ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders three warnings', () => {
    const segment = {
      warnings: [ null, true, true, true ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders no warnings', () => {
    const segment = {
      warnings: [ null, false, false, false ]
    }
    const wrapper = shallow(<Warnings segment={segment} />)
    expect(wrapper).toMatchSnapshot()
  })
})
