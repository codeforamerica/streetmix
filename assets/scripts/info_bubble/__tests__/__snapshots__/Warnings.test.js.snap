// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnings renders no warnings 1`] = `""`;

exports[`Warnings renders three warnings 1`] = `
<div
  className="info-bubble-warnings"
>
  <p
    key="segments.warnings.does-not-fit"
  >
    <FormattedMessage
      defaultMessage="This segment doesn’t fit within the street."
      id="segments.warnings.does-not-fit"
      values={Object {}}
    />
  </p>
  <p
    key="segments.warnings.not-wide"
  >
    <FormattedMessage
      defaultMessage="This segment might not be wide enough."
      id="segments.warnings.not-wide"
      values={Object {}}
    />
  </p>
  <p
    key="segments.warnings.too-wide"
  >
    <FormattedMessage
      defaultMessage="This segment might be too wide."
      id="segments.warnings.too-wide"
      values={Object {}}
    />
  </p>
</div>
`;

exports[`Warnings renders two warnings 1`] = `
<div
  className="info-bubble-warnings"
>
  <p
    key="segments.warnings.does-not-fit"
  >
    <FormattedMessage
      defaultMessage="This segment doesn’t fit within the street."
      id="segments.warnings.does-not-fit"
      values={Object {}}
    />
  </p>
  <p
    key="segments.warnings.too-wide"
  >
    <FormattedMessage
      defaultMessage="This segment might be too wide."
      id="segments.warnings.too-wide"
      values={Object {}}
    />
  </p>
</div>
`;

exports[`Warnings renders warning 1 1`] = `
<div
  className="info-bubble-warnings"
>
  <p
    key="segments.warnings.does-not-fit"
  >
    <FormattedMessage
      defaultMessage="This segment doesn’t fit within the street."
      id="segments.warnings.does-not-fit"
      values={Object {}}
    />
  </p>
</div>
`;

exports[`Warnings renders warning 2 1`] = `
<div
  className="info-bubble-warnings"
>
  <p
    key="segments.warnings.not-wide"
  >
    <FormattedMessage
      defaultMessage="This segment might not be wide enough."
      id="segments.warnings.not-wide"
      values={Object {}}
    />
  </p>
</div>
`;

exports[`Warnings renders warning 3 1`] = `
<div
  className="info-bubble-warnings"
>
  <p
    key="segments.warnings.too-wide"
  >
    <FormattedMessage
      defaultMessage="This segment might be too wide."
      id="segments.warnings.too-wide"
      values={Object {}}
    />
  </p>
</div>
`;
